src/modules/rtp_media_server/rms_media.c

```diff
@@ -0,0 +1,320 @@
+		ms_factory_register_filter(f, rms_ms_filter_descs[i]);
+	}
+	ms_factory_init_plugins(f);
+	ms_factory_enable_statistics(f, TRUE);
+	ms_factory_reset_statistics(f);
+	return f;
+}
+
+int rms_media_init()
+{
+	OrtpMemoryFunctions ortp_memory_functions;
+	ortp_memory_functions.malloc_fun = ptr_shm_malloc;
+	ortp_memory_functions.realloc_fun = ptr_shm_realloc;
+	ortp_memory_functions.free_fun = ptr_shm_free;
+	ortp_set_memory_functions(&ortp_memory_functions);
+	ortp_init();
+	vars = ortp_malloc(sizeof(shared_global_vars_t));
+	return 1;
+}
+
+static MSTicker *rms_create_ticker(char *name)
+{
+	MSTickerParams params;
+	params.name = name;
+	params.prio = MS_TICKER_PRIO_NORMAL;
+	return ms_ticker_new_with_params(&params);
+}
+
+void rms_media_destroy(call_leg_media_t *m)
+{
+	LM_INFO("rtp_session_destroy[%p]\n", m->rtps);
+	rtp_session_destroy(m->rtps);
+	m->rtps = NULL;
+	LM_INFO("ms_ticker[%p]\n", m->ms_ticker);
+	ms_ticker_destroy(m->ms_ticker);
+	m->ms_ticker = NULL;
+	LM_INFO("ms_factory_destroy[%p]\n", m->ms_factory);
+	ms_factory_destroy(m->ms_factory);
+	m->ms_factory = NULL;
+}
+
+int create_call_leg_media(call_leg_media_t *m)
+{
+	m->ms_factory = rms_create_factory();
+	// create caller RTP session
+	LM_INFO("RTP session [%s:%d]<>[%s:%d]\n", m->local_ip.s, m->local_port,
+			m->remote_ip.s, m->remote_port);
+	m->rtps = ms_create_duplex_rtp_session(m->local_ip.s, m->local_port,
+			m->local_port + 1, ms_factory_get_mtu(m->ms_factory));
+	rtp_session_set_remote_addr_full(m->rtps, m->remote_ip.s, m->remote_port,
+			m->remote_ip.s, m->remote_port + 1);
+	rtp_session_set_payload_type(m->rtps, m->pt->type);
+	rtp_session_enable_rtcp(m->rtps, FALSE);
+	// create caller filters : rtprecv1/rtpsend1/encoder1/decoder1
+	m->ms_rtprecv = ms_factory_create_filter(m->ms_factory, MS_RTP_RECV_ID);
+	m->ms_rtpsend = ms_factory_create_filter(m->ms_factory, MS_RTP_SEND_ID);
+
+	LM_INFO("codec[%s]\n", m->pt->mime_type);
+	m->ms_encoder = ms_factory_create_encoder(m->ms_factory, m->pt->mime_type);
+	if(!m->ms_encoder) {
+		LM_ERR("creating encoder failed.\n");
+		return 0;
+	}
+	m->ms_decoder = ms_factory_create_decoder(m->ms_factory, m->pt->mime_type);
+
+	/* set filter params */
+	ms_filter_call_method(m->ms_rtpsend, MS_RTP_SEND_SET_SESSION, m->rtps);
+	ms_filter_call_method(m->ms_rtprecv, MS_RTP_RECV_SET_SESSION, m->rtps);
+	return 1;
+}
+
+int rms_bridge(call_leg_media_t *m1, call_leg_media_t *m2)
+{
+	MSConnectionHelper h;
+	m1->ms_ticker = rms_create_ticker(NULL);
+
+	// direction 1
+	ms_connection_helper_start(&h);
+	ms_connection_helper_link(&h, m1->ms_rtprecv, -1, 0);
+	ms_connection_helper_link(&h, m2->ms_rtpsend, 0, -1);
+
+	// direction 2
+	ms_connection_helper_start(&h);
+	ms_connection_helper_link(&h, m2->ms_rtprecv, -1, 0);
+	ms_connection_helper_link(&h, m1->ms_rtpsend, 0, -1);
+
+	ms_ticker_attach_multiple(
+			m1->ms_ticker, m1->ms_rtprecv, m2->ms_rtprecv, NULL);
+
+	return 1;
+}
+
+#define MS_UNUSED(x) ((void)(x))
```

Comment:
What is the purpose of this define?

Reply:
I modified the includes for mediastreamer2 and included the file that was defining this macro instead.